File: android/app/src/main/java/com/ofermehulal/cocktailsapp/data/models/Recipe.kt
package com.ofermehulal.cocktailsapp.data.models

data class Recipe(
    val id: String = "",
    val name: String = "",
    val description: String = "",
    val baseSpirit: String = "",
    val tastes: List<String> = emptyList(),
    val difficulty: String = "",
    val cost: String = "",
    val imageUrl: String = "",
    val ingredients: List<Ingredient> = emptyList(),
    val instructions: List<String> = emptyList(),
    val garnish: String = "",
    val tips: String = ""
)

data class Ingredient(
    val id: String = "",
    val name: String = "",
    val amount: String = ""
)

// For Firebase queries
data class RecipeFilter(
    val spirit: String? = null,
    val tastes: List<String> = emptyList(),
    val difficulty: String? = null,
    val cost: String? = null
)

data class IngredientFilter(
    val selectedIngredients: Set<String> = emptySet()
)

File: ios/CocktailsApp/CocktailsApp/Models/Recipe.swift
import Foundation
import FirebaseFirestoreSwift

struct Recipe: Identifiable, Codable {
    @DocumentID var id: String?
    let name: String
    let description: String
    let baseSpirit: String
    let tastes: [String]
    let difficulty: String
    let cost: String
    let imageUrl: String
    let ingredients: [Ingredient]
    let instructions: [String]
    let garnish: String
    let tips: String

    enum CodingKeys: String, CodingKey {
        case id
        case name
        case description
        case baseSpirit
        case tastes
        case difficulty
        case cost
        case imageUrl
        case ingredients
        case instructions
        case garnish
        case tips
    }
}

struct Ingredient: Identifiable, Codable {
    let id: String
    let name: String
    let amount: String

    enum CodingKeys: String, CodingKey {
        case id
        case name
        case amount
    }
}

struct RecipeFilter {
    var spirit: String?
    var tastes: [String] = []
    var difficulty: String?
    var cost: String?
}

struct IngredientFilter {
    var selectedIngredients: Set<String> = []
}
